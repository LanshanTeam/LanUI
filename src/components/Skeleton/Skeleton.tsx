import { ReactNode, useEffect } from 'react';
import './style/skeleton.less';

export interface SkeletonProps {
    loading: Boolean;
    height?: string;
    size?: number;
    circle?: Boolean;
    children?: ReactNode;
}

const Skeleton = (props: SkeletonProps) => {
    const { loading, height = '30px', circle, size = 6, children } = props;

    useEffect(() => {
        validateSkeletonProps(props);
    }, []);

    const validateSkeletonProps = (props: SkeletonProps): void => {
        if (props.size !== undefined && props.circle === undefined) {
            throw new Error(
                'If size is provided, circle must also be provided.'
            );
        }
    };

    return loading ? (
        <div
            className="loading"
            style={{
                width: circle ? 5 * size + 'px' : '100%',
                height: circle ? 5 * size + 'px' : height,
                borderRadius: circle ? '50%' : 0,
            }}
        ></div>
    ) : (
        <>{children}</>
    );
};

export default Skeleton;
