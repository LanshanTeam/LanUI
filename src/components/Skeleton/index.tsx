import Skeleton from './Skeleton';
import '../../style/index';
export default Skeleton;
