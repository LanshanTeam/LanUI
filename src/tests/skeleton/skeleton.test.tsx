import React from 'react';
import { render } from '@testing-library/react';
import Skeleton from '../../components/Skeleton/index';
import { SkeletonProps } from '@/components/Skeleton/Skeleton';

const genSkeleton = (props: SkeletonProps) =>
    render(
        <div data-testid="skeleton">
            <Skeleton {...props} />
        </div>
    );

describe('Skeleton Component', () => {
    it('should render with custom height correctly', () => {
        const { getByTestId } = genSkeleton({ loading: true, height: '20px' });
        expect(getByTestId('skeleton')).toMatchSnapshot();
    });

    it('should render animated skeleton correctly', () => {
        const { getByTestId } = genSkeleton({ loading: true, active: true });
        expect(getByTestId('skeleton')).toMatchSnapshot();
    });

    it('should render with rows correctly', () => {
        const { getByTestId } = genSkeleton({ loading: true, rows: 2 });
        expect(getByTestId('skeleton')).toMatchSnapshot();
    });

    it('should render circle skeleton correctly', () => {
        const { getByTestId } = genSkeleton({ loading: true, circle: true });
        expect(getByTestId('skeleton')).toMatchSnapshot();
    });

    it('should render with adaptive height correctly', () => {
        const { getByTestId } = render(
            <div data-testid="skeleton">
                <Skeleton loading={true}>
                    <div>CodingStartup</div>
                    <div>
                        Lorem ipsum dolor, sit amet consectetur adipisicing
                        elit. Ex quasienim facere commodi omnis. . .
                    </div>
                </Skeleton>
            </div>
        );
        expect(getByTestId('skeleton')).toMatchSnapshot();
    });
});
