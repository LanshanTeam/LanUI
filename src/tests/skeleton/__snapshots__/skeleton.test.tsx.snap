// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Component should render animated skeleton correctly 1`] = `
<div
  data-testid="skeleton"
>
  <div
    class="loading active"
    style="width: 100%; height: 30px; border-radius: 0;"
  />
</div>
`;

exports[`Skeleton Component should render circle skeleton correctly 1`] = `
<div
  data-testid="skeleton"
>
  <div
    class="loading "
    style="width: 30px; height: 30px; border-radius: 50%;"
  />
</div>
`;

exports[`Skeleton Component should render with adaptive height correctly 1`] = `
<div
  data-testid="skeleton"
>
  <div
    class="loading "
    style="width: 100%; height: 60px; border-radius: 0;"
  />
</div>
`;

exports[`Skeleton Component should render with custom height correctly 1`] = `
<div
  data-testid="skeleton"
>
  <div
    class="loading "
    style="width: 100%; height: 20px; border-radius: 0;"
  />
</div>
`;

exports[`Skeleton Component should render with rows correctly 1`] = `
<div
  data-testid="skeleton"
>
  <div
    class="loading "
    style="width: 100%; height: 30px; border-radius: 0;"
  />
  <div
    class="loading "
    style="width: 100%; height: 30px; border-radius: 0;"
  />
</div>
`;
